node/core/candidate-validation/src/lib.rs

```diff
@@ -296,30 +302,53 @@ where
 	Context: SubsystemContext<Message = CandidateValidationMessage>,
 	Context: overseer::SubsystemContext<Message = CandidateValidationMessage>,
 {
-	let (validation_data, validation_code) =
-		match find_assumed_validation_data(ctx, &descriptor).await? {
-			AssumptionCheckOutcome::Matches(validation_data, validation_code) =>
-				(validation_data, validation_code),
-			AssumptionCheckOutcome::DoesNotMatch => {
-				// If neither the assumption of the occupied core having the para included or the assumption
-				// of the occupied core timing out are valid, then the persisted_validation_data_hash in the descriptor
-				// is not based on the relay parent and is thus invalid.
-				return Ok(Ok(ValidationResult::Invalid(InvalidCandidate::BadParent)))
-			},
-			AssumptionCheckOutcome::BadRequest =>
-				return Ok(Err(ValidationFailed("Assumption Check: Bad request".into()))),
-		};
+	let validation_data = match find_assumed_validation_data(ctx, &descriptor).await? {
+		AssumptionCheckOutcome::Matches(validation_data) => validation_data,
+		AssumptionCheckOutcome::DoesNotMatch => {
+			// If neither the assumption of the occupied core having the para included or the assumption
+			// of the occupied core timing out are valid, then the persisted_validation_data_hash in the descriptor
+			// is not based on the relay parent and is thus invalid.
+			return Ok(Ok(ValidationResult::Invalid(InvalidCandidate::BadParent)))
+		},
+		AssumptionCheckOutcome::BadRequest =>
+			return Ok(Err(ValidationFailed("Assumption Check: Bad request".into()))),
+	};
 
-	let validation_result = validate_candidate_exhaustive(
-		validation_host,
-		validation_data,
-		validation_code,
+	let mut validation_result = validate_candidate_exhaustive(
+		&mut validation_host.clone(),
+		validation_data.clone(),
+		None,
```

Comment:
Nit: Reading this code I can't help but notice that this function is a misnomer now. It is not exhaustive now since we do not necessarily provide validation code.